validator/remote/src/main/java/tech/pegasys/teku/validator/remote/RemoteValidatorApiHandler.java

```diff
@@ -127,26 +128,49 @@ public SafeFuture<Map<BLSPublicKey, Integer>> getValidatorIndices(
       return SafeFuture.completedFuture(emptyMap());
     }
     return sendRequest(
-        () ->
-            makeBatchedValidatorRequest(publicKeys, ValidatorResponse::getIndex)
-                .orElse(emptyMap()));
+        () -> makeValidatorRequest(publicKeys, ValidatorResponse::getIndex).orElse(emptyMap()));
   }
 
   @Override
   public SafeFuture<Optional<Map<BLSPublicKey, ValidatorStatus>>> getValidatorStatuses(
-      Collection<BLSPublicKey> publicKeys) {
-    return sendRequest(() -> makeBatchedValidatorRequest(publicKeys, ValidatorResponse::getStatus));
+      final Collection<BLSPublicKey> publicKeys) {
+    return sendRequest(() -> makeValidatorRequest(publicKeys, ValidatorResponse::getStatus));
+  }
+
+  private <T> Optional<Map<BLSPublicKey, T>> makeValidatorRequest(
+      final Collection<BLSPublicKey> publicKeys,
+      final Function<ValidatorResponse, T> valueExtractor) {
+    try {
+      return apiClient
+          .postValidators(convertPublicKeysToValidatorIds(publicKeys))
+          .map(responses -> convertToValidatorMap(responses, valueExtractor));
+    } catch (final PostStateValidatorsNotExistingException __) {
+      LOG.debug(
+          "POST method is not available for getting validators from state. Will use GET instead.");
```

Comment:
so here, also remember that there was an exception, and never try again...

Basically each epoch we'd still attempt this post and waste time with the failure if we go this way, where as if we make our first call and that fails, then we move on to gets, the next time we can just do gets... it may just look something like
```
if (isPostWorking) {
    try {
      apiclient.postValidators(....
    } catch () {
      isPostWorking.setAtomicBoolean(false);
      LOG...
    }
}
return makeBatchedValidatorRequest...
```

Reply:
Yeah that makes sense. Will implement.